snuba/datasets/storages/type_condition_enforcer.py

```diff
@@ -0,0 +1,30 @@
+from snuba.clickhouse.processors import QueryProcessor
+from snuba.clickhouse.query import Query
+from snuba.query.conditions import binary_condition, ConditionFunctions
+from snuba.query.expressions import Column, Literal
+from snuba.request.request_settings import RequestSettings
+
+
+class TypeConditionEnforcer(QueryProcessor):
+    """
+    Enforces that transactions are never returned from the events entity.
+    This condition is required for the events storage only, transactions are never
+    present in errors storage. If the query contains any other (likely more restrictive)
+    condition on type or group ID, do not apply the additional condition.
+    """
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        type_cols = {"type", "group_id"}
+
+        cols = {
```

Comment:
This seems like it is missing some edge cases. What if there is a query for `type != 'default'`? Is the idea that if they have any kind of type filter then they get what they get?

Reply:
Good point. I think I'm going to drop the check for a `type` column here, and always add the condition unless there's a condition on `group_id`. I initially wanted to avoid adding the duplicate condition `type != transaction` everywhere (Sentry is still adding it everywhere until this is rolled out) but on second thoughts it probably doesn't matter a great deal if there is a duplicate condition while we are rolling this out.